test/infrastructure/docker/internal/controllers/dockermachine_controller.go

```diff
@@ -353,13 +352,6 @@ func (r *DockerMachineReconciler) reconcileNormal(ctx context.Context, cluster *
 		return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
 	}
 
-	// If the control plane is not yet initialized, there is no API server to contact to get the ProviderID for the Node
```

Comment:
@fabriziopandini I had to remove this change. This broke standalone Machines (and thus the KCP adoption e2e test). 

Essentially for standalone machines we have to set the providerID below to even get to `ControlPlaneInitializedCondition` true. Otherwise we will never get there.